gen.go

```
@@ -123,3 +152,29 @@ Ensure you have the following installed:
 		},
 	}
 )
+
+func checkoutTemplateRef(wt *git.Worktree, templateRef string) error {
+	// Try checking out using the hash first
+	if err := wt.Checkout(&git.CheckoutOptions{
+		Hash: plumbing.NewHash(templateRef),
+	}); err == nil {
+		return nil
+	}
+
+	// Try checking out as a branch name
+	if err := wt.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.NewBranchReferenceName(templateRef),
+	}); err == nil {
+		return nil
+	}
+
+	// Try checking out as a tag
+	if err := wt.Checkout(&git.CheckoutOptions{
+		Branch: plumbing.NewTagReferenceName(templateRef),
+	}); err == nil {
+		return nil
+	}
+
+	// If all attempts fail, return an error
+	return fmt.Errorf("failed to checkout %s: not a valid hash, branch, or tag", templateRef)
+}
```

A tip about commenting (which, I'm in favor of commenting liberally), is that the "why" is more important than the "what". For example you don't need the final comment on line 178, it's obvious what's happening there. Rather than the "Try ...." comments at each Checkout call, I'd put one comment at the top to explain that git-go doesn't provide any way we know of to checkout a "ref", instead we just have to try treating it like a hash then branch then tag and see what works. Since this answers the question an engineer really wants to know here, which is, "why do they try all these 3 and ignore the errors? Isn't there just one call analogous to `git checkout` on the CLI?"

I am glad you're adding comments though, thorough documentation is good 🙂

---

Sounds good! I really appreciate the feedback. I just pushed the final edits based on all the comments.